Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WatchTab/WatchTabContent.tsx

```diff
@@ -0,0 +1,326 @@
+import { WatchDataPayload } from '../../WatchVariableSelector/utils/helpers';
+
+import { WatchTabObjectValue } from './WatchTabObjectValue';
+
+const contentContainer = css`
+  display: flex;
+  flex-flow: column nowrap;
+  height: 100%;
+  width: 100%;
+`;
+
+const toolbar = css`
+  display: flex;
+  flex-flow: row nowrap;
+  flex-shrink: 0;
+  height: 24px;
+  padding: 8px 16px;
+`;
+
+const content = css`
+  display: flex;
+  flex-flow: column nowrap;
+  height: calc(100% - 40px);
+  width: 100%;
+`;
+
+const undefinedValue = css`
+  font-family: Segoe UI;
+  font-size: 12px;
+  font-style: italic;
+  height: 16px;
+  line-height: 16px;
+`;
+
+const watchTableStyles: Partial<IDetailsListStyles> = {
+  root: {
+    height: '100%',
+    selectors: {
+      '& > div[role="grid"]': {
+        height: '100%',
+      },
+    },
+  },
+  contentWrapper: {
+    overflowY: 'auto' as 'auto',
+    // fill remaining space after table header row
+    height: 'calc(100% - 60px)',
+  },
+};
+
+const rowStyles: Partial<IDetailsRowStyles> = {
+  cell: { minHeight: 32, padding: '8px 6px' },
+  checkCell: {
+    height: 32,
+    minHeight: 32,
+    selectors: {
+      '& > div[role="checkbox"]': {
+        height: 32,
+      },
+    },
+  },
+  root: { minHeight: 32 },
+};
+
+const addIcon: IIconProps = {
+  iconName: 'Add',
+};
+
+const removeIcon: IIconProps = {
+  iconName: 'Cancel',
+};
+
+const NameColumnKey = 'column1';
+const ValueColumnKey = 'column2';
+// TODO: update to office-ui-fabric-react@7.170.x to gain access to "flexGrow" column property to distribute proprotional column widths
+// (name column takes up 1/3 of space and value column takes up the remaining 2/3)
+const watchTableColumns: IColumn[] = [
+  {
+    key: NameColumnKey,
+    name: 'Name',
+    fieldName: 'name',
+    minWidth: 100,
+    maxWidth: 600,
+    isResizable: true,
+  },
+  {
+    key: ValueColumnKey,
+    name: 'Value',
+    fieldName: 'value',
+    minWidth: 100,
+    maxWidth: undefined,
+    isResizable: true,
+  },
+];
+
+const watchTableLayout: DetailsListLayoutMode = DetailsListLayoutMode.justified;
+
+// this can be exported and used in other places
+const getValueFromBotTraceScope = (delimitedProperty: string, botTrace: Activity) => {
+  const propertySegments = delimitedProperty.split('.');
+  const value = propertySegments.reduce(
+    (accumulator: object | string | number | boolean | undefined, segment, index) => {
+      // first try to grab the specified property off the root of the bot trace's memory
+      if (index === 0) {
+        return botTrace?.value[segment];
+      }
+      // if we are not on the root, try accessing the next value of the desired property
+      if (typeof accumulator === 'object') {
+        return accumulator[segment];
+      } else {
+        return undefined;
+      }
+    },
+    undefined
+  );
+  return value;
+};
+
+export const WatchTabContent: React.FC<DebugPanelTabHeaderProps> = ({ isActive }) => {
```

Comment:
why is the tab knows it's active or not, isn't the parent deciding to show it or not?

Reply:
This is the pattern that the debug panel tabs use so I am following it. I do not know the reasoning behind the tabs knowing if they are active or not, but I do not think rewriting it should be part of this PR.

Reply:
The active property is sent to all the tabs but only 1 uses it and is controlled at the parent level. The reason for isActive property passed through to one of the tabs is for implementing the unread notification indicator. 
https://github.com/microsoft/BotFramework-Composer/blob/main/Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WebChatLog/WebChatLogItemHeader.tsx#L23